pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -251,19 +251,34 @@ func RunStartBuild(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra
 	if follow {
 		wg.Add(1)
 		go func() {
-			defer wg.Done()
+			// if --wait option is set, then don't wait for logs to finish streaming
+			// but wait for the build to reach its final state
+			if waitForComplete {
+				wg.Done()
+			} else {
+				defer wg.Done()
+			}
 			opts := buildapi.BuildLogOptions{
 				Follow: true,
 				NoWait: false,
 			}
-			rd, err := client.BuildLogs(namespace).Get(newBuild.Name, opts).Stream()
-			if err != nil {
-				fmt.Fprintf(cmd.Out(), "error getting logs: %v\n", err)
-				return
-			}
-			defer rd.Close()
-			if _, err = io.Copy(out, rd); err != nil {
-				fmt.Fprintf(cmd.Out(), "error streaming logs: %v\n", err)
+			for {
+				rd, err := client.BuildLogs(namespace).Get(newBuild.Name, opts).Stream()
+				if err != nil {
+					// if --wait options is set, then retry the connection to build logs
+					// when we hit the timeout.
+					if !waitForComplete && strings.Contains(fmt.Sprintf("%s", err), "timed out") {
```

Comment:
@kargakis that function will not catch the timeout I think since the error type is REST something.